### DISCUSSION TITLE
Add Custom Tags for VM’s on vSphere using openshift-installer cli

### DISCUSSION THREAD

**Discussion by SergeyHudinskiy** [okd-project/okd#1268] [General]
Is it possible for the openshift-installer to add custom tags for the VM's on the vSphere?

As far as I understand we need to use  **install-config.yaml** , but there is no key called **tag** for this for example. 

I've found the following **https://access.redhat.com/solutions/6257541**
but it is not clear how to do this. 

Do you have any ideas?

Thanks
Sergey

**Comment by jcpowermac** [comment]
@SergeyHudinskiy when installing with IPI vCenter tags are created. UPI the user is responsible for creating the tag. 
What is the use-case for "custom tags"?

**Reply by SergeyHudinskiy** [reply]
Hi @jcpowermac, thanks for your reply. Right, with IPI a specific tag is created, but I would like to add an additional custom tag, it is necessary for implementing backup solution on vSphere. So my questions is Is there a way I can add additional tags to VM's on vSphere using openshift-installer or I need to this manually?

**Reply by jcpowermac** [reply]
The tags were only created so we can easily delete vCenter objects if someone wants to destroy the cluster. I don't see us adding the ability to create custom tags since that could easily be done day 2 via govc, powercli or the vCenter UI.

**Reply by JaimeMagiera** [reply]
Right. A simple shell script calling govc could take as input the tag created by the installer, select all objects with said tag, and append additional tags. @SergeyHudinskiy, are you looking to add additional tags to all object types in vSphere, or just VMs, or...?

**Reply by SergeyHudinskiy** [reply]
Hi @JaimeMagiera  I would like to add an additional tag to VM's.

**Reply by SergeyHudinskiy** [reply]
Thank you so much for you quick answers @jcpowermac and @JaimeMagiera !
